### PR TITLE
Clean up Java services, add Docker

### PR DESCRIPTION
**Pull request by robbkidd**
Also updates from WRITE_KEY to API_KEY.

### CODE REVIEW

File: java/frontend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+RUN gradle bootJar --no-daemon
+
+FROM openjdk:13-jdk-alpine
+VOLUME /tmp
+ENV HONEY_OTEL_VER=0.1.1
+ENV HONEY_JAVA_AGENT=honeycomb-opentelemetry-javaagent-${HONEY_OTEL_VER}-all.jar
+ENV JAVA_TOOL_OPTIONS=-javaagent:${HONEY_JAVA_AGENT}
+COPY --from=build /home/gradle/libs/${HONEY_JAVA_AGENT} ./${HONEY_JAVA_AGENT}
+COPY --from=build /home/gradle/build/libs/*.jar app.jar
+CMD ["java", "-jar", "app.jar"]
```
Comment (vreynolds) [inline java/frontend/Dockerfile:18]: Should these be running the javaagent?

Comment (robbkidd) [reply java/frontend/Dockerfile:18]: The agent gets retrieved earlier in the Dockerfile and is set for the java runtime by the `ENV JAVA_TOOL_OPTIONS=` line.

File: java/frontend/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
+FROM gradle:7.0.1-jdk11 AS build
+WORKDIR /home/gradle/
+RUN mkdir ./libs
+ENV HONEY_OTEL_VER=0.1.1
```
Comment (robbkidd) [inline java/frontend/Dockerfile:4]: I reckon I could update the version here.

**Review by vreynolds — APPROVED** [review]

**Review by vreynolds — APPROVED** [review]
whale

**Pull request merged**